pandas/tests/indexing/test_loc.py

```diff
@@ -1644,6 +1644,14 @@ def test_loc_getitem_slice_labels_int_in_object_index(self, frame_or_series, val
         expected = frame_or_series(range(4), index=[value, "first", 2, "third"])
         tm.assert_equal(result, expected)
 
+    def test_loc_getitem_slice_columns_mixed_dtype(self):
+        # GH: 20975
+        df = DataFrame({"test": 1, 1: 2, 2: 3}, index=[0])
+        expected = DataFrame(
+            data=[[2, 3]], index=[0], columns=pd.Index([1, 2], dtype=object)
+        )
+        tm.assert_equal(df.loc[:, 1:], expected)
```

Comment:
does `assert_frame_equal` work?

Reply:
Yes that works and modification made accordingly. Thanks!